### PR TITLE
libvpx-devel:  install vpxdec/vpxenc executables 

### PR DESCRIPTION
**Pull request by i0ntempest**
###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS x.y
Xcode x.y / Command Line Tools x.y.z

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by i0ntempest** [comment]
These executables are required for my new port `av1an`. Also maybe the new `install_name_tool` block can be improved.

**Comment by mascguy** [comment]
Zhenfu, let's update the Devel port only. Once merged - and I've validated that it builds across all macOS releases - I'll update the non-Devel port.

File: multimedia/libvpx-devel/Portfile [hunk 1]
```diff
@@ -159,6 +160,11 @@ post-destroot {
     foreach f [glob ${destroot}${prefix}/lib/${my_name}.*.dylib] {
         system "install_name_tool -id [string map [list ${destroot} ""] ${f}] ${f}"
     }
+    foreach f [glob ${destroot}${prefix}/bin/*] {
+        system "install_name_tool -change \
+        [lindex [split [glob ${destroot}${prefix}/lib/${my_name}.*.dylib] "/"] end] \
+        [string map [list ${destroot} ""] [glob ${destroot}${prefix}/lib/${my_name}.*.dylib]] ${f}"
+    }
 }
```
Comment (mascguy) [inline multimedia/libvpx-devel/Portfile:168]: @jmroot / @ryandesign / Anyone, Zhenfu was wondering whether we could potentially improve this. Thoughts...?

File: multimedia/libvpx-devel/Portfile [hunk 2]
```diff
@@ -159,6 +160,11 @@ post-destroot {
     foreach f [glob ${destroot}${prefix}/lib/${my_name}.*.dylib] {
         system "install_name_tool -id [string map [list ${destroot} ""] ${f}] ${f}"
     }
+    foreach f [glob ${destroot}${prefix}/bin/*] {
+        system "install_name_tool -change \
+        [lindex [split [glob ${destroot}${prefix}/lib/${my_name}.*.dylib] "/"] end] \
+        [string map [list ${destroot} ""] [glob ${destroot}${prefix}/lib/${my_name}.*.dylib]] ${f}"
+    }
```
Comment (mascguy) [inline multimedia/libvpx-devel/Portfile:167]: @jmroot / @ryandesign / Anyone, Zhenfu was wondering whether we could potentially improve this. Thoughts...?

File: multimedia/libvpx-devel/Portfile [hunk 3]
```diff
@@ -51,14 +51,15 @@ compiler.blacklist-append  {*gcc-[3-4].*} {clang < 900} {macports-clang-3.*}
 # doesn't install docs or examples correctly, so disable them.
 configure.args      --enable-vp8 \
                     --enable-vp9 \
+                    --enable-vp9-highbitdepth \
                     --enable-internal-stats \
                     --enable-pic \
                     --enable-postproc \
                     --enable-multithread \
                     --enable-experimental \
+                    --enable-examples \
```
Comment (mascguy) [inline multimedia/libvpx-devel/Portfile:60]: I'm fine with adding examples to the mix, but you'll need to explicitly destroot them:

```
MacPro2012:work chrisadmin$ gfind libvpx-1.12.0/examples -type f -perm 755 -print | gsort
libvpx-1.12.0/examples/decode_to_md5
libvpx-1.12.0/examples/decode_with_drops
libvpx-1.12.0/examples/postproc
libvpx-1.12.0/examples/set_maps
libvpx-1.12.0/examples/simple_decoder
libvpx-1.12.0/examples/simple_encoder
libvpx-1.12.0/examples/twopass_encoder
libvpx-1.12.0/examples/vp8cx_set_ref
libvpx-1.12.0/examples/vp9_lossless_encoder
libvpx-1.12.0/examples/vp9_spatial_svc_encoder
libvpx-1.12.0/examples/vp9cx_set_ref
libvpx-1.12.0/examples/vpx_temporal_svc_encoder
```

I'd recommend they be placed in directory `${prefix}/share/${my_name}/examples/`.

Comment (mascguy) [reply multimedia/libvpx-devel/Portfile:60]: > I'm fine with adding examples to the mix, but you'll need to explicitly destroot them

Unless there's a Makefile option to include them in the installation, that is. Haven't looked too closely, but perhaps there is one...?

Comment (i0ntempest) [reply multimedia/libvpx-devel/Portfile:60]: I only enabled examples so vpxenc/vpxdec executables are built/installed, do you want all of these executables to be installed? All of these need to be fixed with install_name_tool if so.

Comment (mascguy) [reply multimedia/libvpx-devel/Portfile:60]: > I only enabled examples so vpxenc/vpxdec executables are built/installed, do you want all of these executables to be installed? All of these need to be fixed with install_name_tool if so.

Gotcha, didn't realize those were part of the examples. Never Mind!  :-D

**Review by mascguy — APPROVED** [review]
LGTM. Let's leave this open for a few days, to let folks provide feedback on the `install_name_tool` block.

**Comment by mascguy** [comment]
@Gcenx Dean, I added your updated configure patch, per our earlier discussion: https://github.com/macports/macports-ports/commit/86a9dcf3cabb29908fc657458912e0eb012911ee#commitcomment-77466333

**Pull request merged**